### ISSUE TITLE
Sky region artifacts when using KITTI pretrained weights on KITTI 2015 testing set

### ISSUE THREAD

**Issue by Albertcqq** [gangweix/BANet#13]
Hi @gangweix,

Thank you for the excellent work on BANet! I'm testing BANet-2D on the KITTI 2015 testing set using the provided KITTI pretrained weights (`kitti.pth`), and I noticed some disparity artifacts in the sky/tree canopy regions at the top of the images.

**Environment:**
- GPU: NVIDIA RTX 3060 Laptop (6GB)
- PyTorch: 2.1.0 + CUDA 12.1
- timm: 0.5.4

**Steps to reproduce:**
```bash
python save_disp.py \
    --restore_ckpt ./pretrained_models/kitti.pth \
    -l "/path/to/kitti/2015/testing/image_2/*_10.png" \
    -r "/path/to/kitti/2015/testing/image_3/*_10.png" \
    --output_directory ./output
```

**Issue:**
The sky and tree canopy regions show abnormally high disparity values (red artifacts in JET colormap visualization), while the rest of the disparity map looks correct and high-quality. See the comparison below:

- Paper result (from Fig.1): sky region is clean with near-zero disparity ✅
- My result: sky region has red/yellow artifacts with high disparity values ❌

I have already tried changing `InputPadder` mode from `sintel` (default) to `kitti` (padding only at bottom), which reduced the artifacts but did not fully eliminate them.

**Questions:**
1. Is there any specific post-processing applied to the results shown in the paper (e.g., sky masking, disparity clamping, or median filtering)?
2. Should I use different `InputPadder` settings for KITTI inference?
3. Do you have any suggestions for handling textureless regions (sky, uniform walls) in practical deployment scenarios?

Any advice would be greatly appreciated. Thanks again for open-sourcing this great work!

<img width="1242" height="1125" alt="Image" src="https://github.com/user-attachments/assets/df41bd3e-8330-4311-8390-cfe477b8ba2c" />

<img width="2592" height="288" alt="Image" src="https://github.com/user-attachments/assets/30bf16e7-e64c-44f9-a7dd-c3cd9d0c283e" />
